### DISCUSSION TITLE
How do you prevent a component from stealing focus?

### DISCUSSION THREAD

**Discussion by AdagioB** [quasarframework/quasar#9840] [General - Components / Directives / etc]
I have an input field (vue-autonumeric component) that keeps losing focus due to another component showing up, which I really want to avoid. 
I havea q-popup-proxy that pops up when the data is not correct. 

```
<q-popup-proxy v-if="localValidation"
   transition-show="flip-up"
   transition-hide="flip-down"
    :value="showMessage"
    :offset="[-10, -3]"
   @input="KeepAlive"
   @show="SetFocusToFieldObject">

   <q-banner class="text-white validationMessageBackground" dense rounded>

      <div class="validationMessageTextSize">
         {{ validationMessage }}
      </div>

   </q-banner>
</q-popup-proxy>
```

I have tried to have my code automatically set focus again, when one of those pops up, but I'm having a hard time getting this to work with the vue-autonumeric component. It works fine with q-input component, but that component wasn't useful in my situation. Funny thing is that document.activeElement actually tells me the vue-autonumeric component has focus, but it doesn't act like it (pressing any keys doesn't do anything)
This made me think if there is a way to stop the other components from gaining focus when they become visible. So far I have come up empty

**Comment by AdagioB** [comment]
It seems like I might get the same result by changing it to a q-menu, so that's probably the way I will go, but would still love to hear if there is an option to prevent a component from taking focus on show

**Reply by metalsadman** [reply]
you probably need to await a vue `nextTick`. https://v3.vuejs.org/api/global-api.html#nexttick

**Reply by justinbeaty** [reply]
@AdagioB have you tried using the `no-focus` prop of `q-menu` ?

**Reply by AdagioB** [reply]
Thanks to both of you. I ended up going with using the q-menu with no-focus
